bot/cogs/moderation/silence.py

```diff
@@ -83,18 +103,23 @@ async def silence(self, ctx: Context, duration: HushDurationConverter = 10) -> N
         Duration is capped at 15 minutes, passing forever makes the silence indefinite.
         Indefinitely silenced channels get added to a notifier which posts notices every 15 minutes from the start.
         """
-        await self._get_instance_vars_event.wait()
+        await self._init_task
         log.debug(f"{ctx.author} is silencing channel #{ctx.channel}.")
+
         if not await self._silence(ctx.channel, persistent=(duration is None), duration=duration):
-            await ctx.send(f"{Emojis.cross_mark} current channel is already silenced.")
+            await ctx.send(MSG_SILENCE_FAIL)
             return
+
         if duration is None:
-            await ctx.send(f"{Emojis.check_mark} silenced current channel indefinitely.")
+            await ctx.send(MSG_SILENCE_PERMANENT)
+            await self.unsilence_timestamps.set(ctx.channel.id, -1)
             return
 
-        await ctx.send(f"{Emojis.check_mark} silenced current channel for {duration} minute(s).")
+        await ctx.send(MSG_SILENCE_SUCCESS.format(duration=duration))
 
         self.scheduler.schedule_later(duration * 60, ctx.channel.id, ctx.invoke(self.unsilence))
+        unsilence_time = (datetime.now(tz=timezone.utc) + timedelta(minutes=duration))
```

Comment:
Any reason for the parentheses here? Can also leave out the tz since it only gets converted to the timestamp

Reply:
See the commit message

> `datetime.timestamp()` assumes naïve `datetime`s are in local time, so
getting POSIX timestamps in UTC isn't easy for naïve ones. Technically,
the timestamp's timezone doesn't matter if all code is on the same page
and parsing it with the same timezone. Keeping it in the local timezone
would be okay then, but I feel safer locking it to UTC explicitly.

> Any reason for the parentheses here?

No, they can be removed.

Reply:
I'm not sure I understand the implications of the different computation methods here, but the timestamp is inherently timezone agnostic and generated on the same machine so they should be fine no? In any case it's not a lot of additional code so I'm not against it, but don't entirely understand the reason for it

Reply:
That's incorrect. A timestamp is _not_ timezone agnostic. For naïve datetimes, the timestamp will be relative to the system's timezone. I did point out that it's okay since it's on the same machine, but I preferred the explicitness.

Reply:
I was always surprised by the parens. Do you think it reads better with them?

Reply:
No. This code just went through several iterations. There used to be a function call somewhere in between.